include/oneapi/dpl/pstl/hetero/numeric_impl_hetero.h

```diff
@@ -108,19 +121,210 @@ __pattern_transform_reduce(_ExecutionPolicy&& __exec, _ForwardIterator __first,
     return __res;
 }
 
+template< bool _Inclusive>
+struct __single_group_scan
+{
+    template<typename _Policy, typename _InRng, typename _OutRng, typename _InitType, typename _BinaryOperation, typename _UnaryOp>
+    static void apply(_Policy const & __policy, _InRng __in, _OutRng __out, ::std::size_t __n, _InitType __init, const _BinaryOperation& __bin_op, const _UnaryOp& __unary_op, ::std::uint32_t __wg_size)
+    {
+        using _RangeValueType = decltype(*__in.begin());
+        using _ValueType = decltype(__bin_op(std::declval<_RangeValueType>(), std::declval<_RangeValueType>()));
+
+        using _CustomName = typename _Policy::kernel_name;
+        using _GroupScanKernel =
+                     oneapi::dpl::__par_backend_hetero::__internal::__kernel_name_generator<__par_backend_hetero::__scan_single_wg_dynamic_kernel, _CustomName, _BinaryOperation, _InRng, _OutRng>;
+
+        ::uint32_t __elems_per_item = __par_backend_hetero::__ceiling_div(__n, __wg_size);
+        ::uint32_t __elems_per_wg = __elems_per_item * __wg_size;
+
+        auto __event = __policy.queue().submit([&](sycl::handler& __hdl) {
+            oneapi::dpl::__ranges::__require_access(__hdl, __in, __out);
+
+            auto __lacc = sycl::accessor<_ValueType, 1, sycl::access_mode::read_write, sycl::target::local>(sycl::range<1>{__elems_per_wg}, __hdl);
+
+            __hdl.parallel_for<_GroupScanKernel>(sycl::nd_range<1>(__wg_size, __wg_size), [=](sycl::nd_item<1> __self_item) {
+                const auto& __group = __self_item.get_group();
+                const auto __item_id = __self_item.get_local_linear_id();
+
+                for (uint16_t __i = 0; __i < __elems_per_item; ++__i)
```

Comment:
`__elems_per_item `is `uint32_t`, but a loop counter is `uint16_t`

Reply:
Will change all constexpr types here to `uint16_t`.